Mathlib/Algebra/Group/Int.lean

```diff
@@ -95,6 +96,106 @@ lemma toAdd_zpow (a : Multiplicative ℤ) (b : ℤ) : toAdd (a ^ b) = toAdd a *
 
 end Multiplicative
 
+/-! ### Units -/
+
+variable {u v : ℤ}
+
+lemma units_natAbs (u : ℤˣ) : natAbs u = 1 :=
+  Units.ext_iff.1 <|
+    Nat.units_eq_one
+      ⟨natAbs u, natAbs ↑u⁻¹, by rw [← natAbs_mul, Units.mul_inv]; rfl, by
+        rw [← natAbs_mul, Units.inv_mul]; rfl⟩
+#align int.units_nat_abs Int.units_natAbs
+
+@[simp] lemma natAbs_of_isUnit (hu : IsUnit u) : natAbs u = 1 := units_natAbs hu.unit
+
+lemma isUnit_eq_one_or (hu : IsUnit u) : u = 1 ∨ u = -1 := by
+  simpa only [natAbs_of_isUnit hu] using natAbs_eq u
+#align int.is_unit_eq_one_or Int.isUnit_eq_one_or
+
+-- theorem units_eq_one_or (u : ℤˣ) : u = 1 ∨ u = -1 := by
+--   simpa only [Units.ext_iff, units_natAbs] using natAbs_eq u
+-- #align int.units_eq_one_or Int.units_eq_one_or
+
+-- theorem units_ne_iff_eq_neg {u u' : ℤˣ} : u ≠ u' ↔ u = -u' := by
+--   rcases units_eq_one_or u with rfl | rfl <;>
+--   rcases units_eq_one_or u' with rfl | rfl <;>
+--   decide
+
```

Comment:
```suggestion
```
if these really can't go here, but I'd like to understand why not

Reply:
The `Neg ℤˣ` instance requires the `Ring ℤ` instance. In contrast, talking about elements of `ℤ` respecting `IsUnit` only requires the `CommMonoid ℤ` and `Neg ℤ` instances.